orbit/pkg/update/hash.go

```diff
@@ -12,29 +12,34 @@ import (
 	"github.com/theupdateframework/go-tuf/data"
 )
 
-// CheckFileHash checks the file at the local path against the provided hash
-// functions.
-func CheckFileHash(meta *data.TargetFileMeta, localPath string) error {
-	hashFunc, hashVal, err := selectHashFunction(meta)
+// checkFileHash checks the file at the local path against the provided hash functions.
+func checkFileHash(meta *data.TargetFileMeta, localPath string) error {
+	localHash, metaHash, err := fileHashes(meta, localPath)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to calculate local file hash: %s", err)
 	}
+	if !bytes.Equal(localHash, metaHash) {
+		return fmt.Errorf("hash %s does not match expected: %s", data.HexBytes(localHash), data.HexBytes(metaHash))
+	}
+	return nil
+}
 
-	f, err := os.Open(localPath)
+func fileHashes(meta *data.TargetFileMeta, localPath string) (localHash []byte, metaHash []byte, err error) {
```

Comment:
It reads a bit weird/error-prone to have `meta, local` in the input arguments, but `local, meta` in the return values?